irksome/imex.py

```diff
@@ -0,0 +1,267 @@
+    The former is used to move the solution forward in time,
+    while the latter is used both to start the method (filling up
+    the initial stage values) and can be used at each time step
+    to increase the accuracy/stability.  In the limit as
+    the iterator is applied many times per time step,
+    one expects convergence to the solution that would have been
+    obtained from fully-implicit RadauIIA method.
+
+    :arg F: A :class:`ufl.Form` instance describing the implicit part
+            of the semi-discrete problem
+            F(t, u; v) == 0, where `u` is the unknown
+            :class:`firedrake.Function and `v` is the
+            :class:firedrake.TestFunction`.
+    :arg Fexp A :class:`ufl.Form` instance describing the part of the
+            PDE that is explicitly split off.
+    :arg butcher_tableau: A :class:`ButcherTableau` instance giving
+            the Runge-Kutta method to be used for time marching.
+    :arg t: A :class:`firedrake.Constant` instance that always
+            contains the time value at the beginning of a time step
+    :arg dt: A :class:`firedrake.Constant` containing the size of the
+            current time step.  The user may adjust this value between
+            time steps.
+    :arg u0: A :class:`firedrake.Function` containing the current
+            state of the problem to be solved.
+    :arg bcs: An iterable of :class:`firedrake.DirichletBC` containing
+            the strongly-enforced boundary conditions.  Irksome will
+            manipulate these to obtain boundary conditions for each
+            stage of the RK method.
+    :arg it_solver_parameters: A :class:`dict` of solver parameters that
+            will be used in solving the algebraic problem associated
+            with the iterator.
+    :arg prop_solver_parameters: A :class:`dict` of solver parameters that
+            will be used in solving the algebraic problem associated
+            with the propagator.
+    :arg splitting: An callable used to factor the Butcher matrix,
+            currently, only AI is supported.
+    :arg appctx: An optional :class:`dict` containing application context.
+    """
+    def __init__(self, F, Fexp, butcher_tableau,
+                 t, dt, u0, bcs=None,
+                 it_solver_parameters=None,
+                 prop_solver_parameters=None,
+                 splitting=AI,
+                 appctx=None,
+                 nullspace=None):
+        assert isinstance(butcher_tableau, RadauIIA)
+
+        self.u0 = u0
+        self.t = t
+        self.dt = dt
+        self.num_fields = len(u0.function_space())
+        self.num_stages = len(butcher_tableau.b)
+        self.butcher_tableau = butcher_tableau
+
+        # Since this assumes RadauIIA, we drop
+        # the update information on the floor.
+        Fbig, _, UU, bigBCs, gblah, nsp = getFormStage(
+            F, butcher_tableau, u0, t, dt, bcs,
+            splitting=splitting)
```

Comment:
Shouldn't `nullspace` be passed along here?

Reply:
Good catch.  I wonder if this is why some of my test cases have been failing?